pluma/pluma-document-saver.c

```diff
@@ -49,334 +49,334 @@
-
-	g_object_class_install_property (object_class,
-					 PROP_ENCODING,
-					 g_param_spec_boxed ("encoding",
-							     "URI",
-							     "The encoding of the saved file",
-							     PLUMA_TYPE_ENCODING,
-							     G_PARAM_READWRITE |
-							     G_PARAM_CONSTRUCT_ONLY |
-							     G_PARAM_STATIC_STRINGS));
-
-	g_object_class_install_property (object_class,
-					 PROP_NEWLINE_TYPE,
-					 g_param_spec_enum ("newline-type",
-					                    "Newline type",
-					                    "The accepted types of line ending",
-					                    PLUMA_TYPE_DOCUMENT_NEWLINE_TYPE,
-					                    PLUMA_DOCUMENT_NEWLINE_TYPE_LF,
-					                    G_PARAM_READWRITE |
-					                    G_PARAM_STATIC_NAME |
-					                    G_PARAM_STATIC_BLURB |
-					                    G_PARAM_CONSTRUCT_ONLY));
-
-	g_object_class_install_property (object_class,
-					 PROP_FLAGS,
-					 g_param_spec_flags ("flags",
-							     "Flags",
-							     "The flags for the saving operation",
-							     PLUMA_TYPE_DOCUMENT_SAVE_FLAGS,
-							     0,
-							     G_PARAM_READWRITE |
-							     G_PARAM_CONSTRUCT_ONLY));
-
-	signals[SAVING] =
-		g_signal_new ("saving",
-			      G_OBJECT_CLASS_TYPE (object_class),
-			      G_SIGNAL_RUN_LAST,
-			      G_STRUCT_OFFSET (PlumaDocumentSaverClass, saving),
-			      NULL, NULL,
-			      pluma_marshal_VOID__BOOLEAN_POINTER,
-			      G_TYPE_NONE,
-			      2,
-			      G_TYPE_BOOLEAN,
-			      G_TYPE_POINTER);
+    GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+    object_class->dispose = pluma_document_saver_dispose;
+    object_class->finalize = pluma_document_saver_finalize;
+    object_class->set_property = pluma_document_saver_set_property;
+    object_class->get_property = pluma_document_saver_get_property;
+
+    g_object_class_install_property (object_class,
+                                     PROP_DOCUMENT,
+                                     g_param_spec_object ("document",
+                                     "Document",
+                                     "The PlumaDocument this PlumaDocumentSaver is associated with",
+                                     PLUMA_TYPE_DOCUMENT,
+                                     G_PARAM_READWRITE |
+                                     G_PARAM_CONSTRUCT_ONLY |
+                                     G_PARAM_STATIC_STRINGS));
+
+    g_object_class_install_property (object_class,
+                                     PROP_URI,
+                                     g_param_spec_string ("uri",
+                                     "URI",
+                                     "The URI this PlumaDocumentSaver saves the document to",
+                                     "",
+                                     G_PARAM_READWRITE |
+                                     G_PARAM_CONSTRUCT_ONLY |
+                                     G_PARAM_STATIC_STRINGS));
+
+    g_object_class_install_property (object_class,
```

Comment:
Can you please re-check line 290 to 310 (2 blocks) ?
Rest looks Ok.
Thanks

Reply:
should be fine now.